owncloud-client:  4.2.0 -> 5.0.0

hellwolf opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hellwolf commented:
@ofborg build owncloud-client

hellwolf commented:
@ofborg build owncloud-client

hellwolf commented:
@ofborg build owncloud-client

hellwolf commented:
1 package built:
owncloud-client

Result of `nixpkgs-review pr 259634` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>owncloud-client</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2764

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/default.nix:
```diff
@@ -1,34 +1,51 @@
+  version = "5.0.0";
 
-  libregraph = callPackage ./libre-graph-api-cpp-qt-client.nix { };
+  # track src/libsync/CMakeLists.txt
```
Why are those libraries not available for other applications in top-level?

SuperSandro2000 replied:
The library lives in an extra repo and is properly packaged, so it should gets it's own top-level entry.

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/default.nix:
```diff
@@ -1,34 +1,51 @@
+    qt6Packages.qtkeychain
+    libre-graph-api
+    kdsingleapplication
+  ] ++ (lib.optionals stdenv.isDarwin [
```
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/default.nix:
```diff
@@ -1,34 +1,51 @@
+    kdsingleapplication
+  ] ++ (lib.optionals stdenv.isDarwin [
+    libinotify-kqueue sparkleshare
+  ]);
```
```suggestion
  ];
```
Those are redundant

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/kdsingleapplication.nix:
```diff
@@ -0,0 +1,24 @@
+  buildInputs = [ qtbase ];
+
+  dontWrapQtApps = true;
+  cmakeFlags = [ "-DKDSingleApplication_QT6=true" ];
```
Missing meta

hellwolf replied:
ditto re meta of internal library.

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/libre-graph-api-cpp-qt-client.nix:
```diff
@@ -19,17 +18,9 @@ stdenv.mkDerivation rec {
+  nativeBuildInputs = [ cmake ];
   buildInputs = [ qtbase ];
 
+  dontWrapQtApps = true;
```
If there is no wrapper, so we even need to set this?

hellwolf replied:
I just tried, it seems not needed.

SuperSandro2000 commented on pkgs/applications/networking/owncloud-client/libre-graph-api-cpp-qt-client.nix:
```diff
@@ -19,17 +18,9 @@ stdenv.mkDerivation rec {
+  dontWrapQtApps = true;
   cmakeFlags = [  ];
-
-  meta = with lib; {
```
Why delete meta?

hellwolf replied:
if it's internal library, is meta relevant?

SuperSandro2000 replied:
Yes, meta is very relevant and everything that uses mkDerivation should have meta.

hellwolf commented:
Seeking some clarification re "internal libraries" from more people.

By internal library, I mean the libraries that for all its intended purpose and likelihood, only the said package will be using them.

In case of this package, there are two of them: 

1. pkgs/applications/networking/owncloud-client/kdsingleapplication.nix
2. pkgs/applications/networking/owncloud-client/libre-graph-api-cpp-qt-client.nix

hellwolf commented:
I moved all three packages to pkgs/by-name

hellwolf commented:
3 packages built:
kdsingleapplication libre-graph-api-cpp-qt-client owncloud-client

Result of `nixpkgs-review pr 259634` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>kdsingleapplication</li>
    <li>libre-graph-api-cpp-qt-client</li>
    <li>owncloud-client</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/kd/kdsingleapplication/package.nix:
```diff
@@ -0,0 +1,33 @@
+    maintainers = with maintainers; [ hellwolf ];
+    platforms = platforms.all;
+    license = licenses.mit;
+    changelog = "https://github.com/KDAB/${pname}/releases/tag/v${version}";
```
```suggestion
    changelog = "https://github.com/KDAB/KDSingleApplication/releases/tag/v${version}";
```
it's really nice when reviewers can click on links

SuperSandro2000 commented on pkgs/by-name/kd/kdsingleapplication/package.nix:
```diff
@@ -0,0 +1,33 @@
+    description = " KDAB's helper class for single-instance policy applications";
+    homepage = "https://www.kdab.com/";
+    maintainers = with maintainers; [ hellwolf ];
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```
qt things are usually only at most unix

hellwolf replied:
done

SuperSandro2000 commented on pkgs/by-name/li/libre-graph-api-cpp-qt-client/package.nix:
```diff
@@ -19,17 +18,16 @@ stdenv.mkDerivation rec {
+    platforms = platforms.all;
     license = licenses.asl20;
-    changelog = "https://github.com/owncloud/libre-graph-api-cpp-qt-client/releases/tag/v${version}";
+    changelog = "https://github.com/owncloud/${pname}/releases/tag/v${version}";
```
Please revert this line

hellwolf replied:
done

SuperSandro2000 commented on pkgs/by-name/ow/owncloud-client/package.nix:
```diff
@@ -0,0 +1,60 @@
+    qt6.qtsvg # Needed for the systray icon
+    qt6Packages.qtkeychain
+    libre-graph-api-cpp-qt-client
+    kdsingleapplication # new dependency since v5.0.0
```
```suggestion
    kdsingleapplication
```
that is obvious from the git history, no need to comment

hellwolf replied:
done

SuperSandro2000 commented on pkgs/by-name/ow/owncloud-client/package.nix:
```diff
@@ -0,0 +1,60 @@
+    description = "Synchronise your ownCloud with your computer using this desktop client";
+    homepage = "https://owncloud.org";
+    maintainers = with maintainers; [ qknight hellwolf ];
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```

hellwolf replied:
done

SuperSandro2000 reviewed (commented):
some minor adjustments, otherwise LGTM, didn't try

hellwolf commented:
> some minor adjustments, otherwise LGTM, didn't try

done

hellwolf commented:
https://github.com/ofborg build owncloud-client

pbsds commented on pkgs/by-name/li/libre-graph-api-cpp-qt-client/package.nix:
```diff
@@ -19,15 +18,14 @@ stdenv.mkDerivation rec {
     description = "C++ Qt API for Libre Graph, a free API for cloud collaboration inspired by the MS Graph API";
     homepage = "https://owncloud.org";
-    maintainers = with maintainers; [ qknight hellwolf ];
+    maintainers = with maintainers; [ hellwolf ];
```
Why remove qknight?

hellwolf replied:
This is the new package required by owncound after I started to do more packaging. @qknight is still the maintainer of owncloud.

In fact these 3 packages are so tightly coupled that, at the moment it doesn't make a difference since qknight is still in the maintainer list of owncloud-client package itself.

pbsds replied:
checked the `git blame`, looks good

pbsds commented on pkgs/by-name/ow/owncloud-client/package.nix:
```diff
@@ -1,46 +1,59 @@
 , stdenv
 , fetchFromGitHub
-, mkDerivation
+, callPackage
```
```suggestion
```

hellwolf commented:
Thank you @pbsds , please have a look again.

pbsds commented on pkgs/by-name/kd/kdsingleapplication/package.nix:
Please `init at` this file in a separate commit

hellwolf replied:
Okay, I will look this weekend. It takes me time, just volunteering here.

hellwolf replied:
ah just separate commit, that's easy. I thought you wanted new PRs. One sec then.

hellwolf replied:
done!

pbsds reviewed (changes requested):
One nit, then LGTM. I don't have a owncloud setup to test on however.

hellwolf committed:
libre-graph-api-cpp-qt-client: refactor

- move to pkgs/by-name

hellwolf committed:
kdsingleapplication: init at 1.0.0

- new library required by owncloud-client 5.0.0

hellwolf committed:
owncloud-client: 4.2.0 -> 5.0.0

- 5.0.0 adds new a new local dependency: kdsingleapplication
- 5.0.0 starts to use Qt6
- Qt6 requires qt6.qtsvg for tray icon support
- move all packages under by-name

pbsds reviewed (approved):
Diff LGTM

K900 commented on pkgs/by-name/ow/owncloud-client/package.nix:
```diff
@@ -1,46 +1,58 @@
+    libinotify-kqueue sparkleshare
   ];
 
+  wrapQtAppsHook = true;
```
This shouldn't be used?

hellwolf replied:
done
